File: shared/python/aeos_shared/models/incident.py
"""Incident and multimodal input data models."""

from __future__ import annotations

from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, field_validator


class MultimodalInputFormat(str, Enum):
    """Supported multimodal input formats."""

    text = "text"
    json = "json"
    pdf = "pdf"
    image = "image"
    log = "log"
    audio = "audio"
    transcript = "transcript"


class MultimodalInputStatus(str, Enum):
    """Processing status for multimodal inputs."""

    pending = "pending"
    processing = "processing"
    ready = "ready"
    failed = "failed"


class IncidentSeverity(str, Enum):
    """Incident severity levels."""

    critical = "critical"
    high = "high"
    medium = "medium"
    low = "low"


class IncidentStatus(str, Enum):
    """Incident lifecycle status."""

    open = "open"
    in_progress = "in_progress"
    resolved = "resolved"
    escalated = "escalated"


class MultimodalInput(BaseModel):
    """Represents a multimodal input submitted for incident processing."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    format: MultimodalInputFormat
    file_path: str | None = None
    raw_content: str | None = None
    extracted_text: str | None = None
    transcript: str | None = None
    file_size_bytes: int | None = None
    processing_status: MultimodalInputStatus
    created_at: datetime


class Incident(BaseModel):
    """Represents a classified operational incident."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    root_signature: str
    severity: IncidentSeverity | None = None
    confidence_score: float | None = Field(default=None, ge=0.0, le=1.0)
    status: IncidentStatus
    source_input_ref: UUID | None = None
    workflow_id: UUID | None = None
    created_at: datetime
    updated_at: datetime


class ClassificationResult(BaseModel):
    """Result of incident classification by the Incident Analysis Agent."""

    model_config = ConfigDict(from_attributes=True)

    severity: IncidentSeverity | None = None
    confidence_score: float = Field(ge=0.0, le=1.0)
    root_signature: str
    requires_escalation: bool

File: shared/typescript/src/incident.ts
export type MultimodalInputFormat = 'text' | 'json' | 'pdf' | 'image' | 'log' | 'audio' | 'transcript';
export type MultimodalInputStatus = 'pending' | 'processing' | 'ready' | 'failed';
export type IncidentSeverity = 'critical' | 'high' | 'medium' | 'low';
export type IncidentStatus = 'open' | 'in_progress' | 'resolved' | 'escalated';

export interface MultimodalInput {
  id: string;
  format: MultimodalInputFormat;
  file_path?: string;
  raw_content?: string;
  extracted_text?: string;
  transcript?: string;
  file_size_bytes?: number;
  processing_status: MultimodalInputStatus;
  created_at: string;
}

export interface Incident {
  id: string;
  root_signature: string;
  severity?: IncidentSeverity;
  confidence_score?: number;
  status: IncidentStatus;
  source_input_ref?: string;
  workflow_id?: string;
  created_at: string;
  updated_at: string;
}

export interface ClassificationResult {
  severity?: IncidentSeverity;
  confidence_score: number;
  root_signature: string;
  requires_escalation: boolean;
}
